### PR TITLE
[C-3060] Fix mobile outbound link experience

### PR DESCRIPTION
**Pull request by dylanjeffers**
### Description

- Fixes issue where the touch targets for confirming/cancelling an outbound link in the LeavingAudiusDrawer are way too small
- Fixes issue where allowListed external links still open the LeavingAudiusDrawer
- Fixes issues with drawer header styles.

### CODE REVIEW

File: packages/common/src/utils/linking.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+export const externalLinkAllowList = new Set([
+  'facebook.com',
+  'instagram.com',
+  'tiktok.com',
+  'twitter.com',
+  'x.com',
+  'blog.audius.co'
+])
+
+export const isAllowedExternalLink = (link: string) => {
```
Comment (dylanjeffers) [inline packages/common/src/utils/linking.ts:10]: lmk if folks can think of better name. "known" possibly?

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/dj-c-3060-fix-outbound-link-experienc

File: packages/mobile/src/components/action-drawer/ActionDrawer.tsx [hunk 2]
```diff
@@ -4,11 +4,13 @@ import { useCallback } from 'react'
 import type { Modals } from '@audius/common'
 import type { TextStyle, ViewStyle } from 'react-native'
 import { TouchableHighlight, View } from 'react-native'
+import type { SetOptional } from 'type-fest'
```
Comment (amendelsohn) [inline packages/mobile/src/components/action-drawer/ActionDrawer.tsx:7]: ooh cool

File: packages/mobile/src/components/core/Text.tsx [hunk 3]
```diff
@@ -68,7 +68,13 @@ export const Text = (props: TextProps) => {
...
         marginTop:
           weight === 'demiBold' && Platform.OS === 'ios'
-            ? spacing(fontSize === 'large' ? 1 : fontSize === 'small' ? 0.5 : 0)
+            ? spacing(
+                fontSize && ['xl', 'large'].includes(fontSize)
+                  ? 1
+                  : fontSize === 'small'
+                  ? 0.5
+                  : 0
+              )
```
Comment (amendelsohn) [inline packages/mobile/src/components/core/Text.tsx:77]: size -> spacing map maybe better

File: packages/mobile/src/components/feed-filter-drawer/FeedFilterDrawer.tsx [hunk 4]
```diff
@@ -74,7 +61,11 @@ export const FeedFilterDrawer = () => {
   return (
     <ActionDrawer
       modalName={MODAL_NAME}
-      renderTitle={() => <Text style={styles.title}>{messages.title}</Text>}
+      title={
+        <Text color='neutral' textTransform='none'>
```
Comment (amendelsohn) [inline packages/mobile/src/components/feed-filter-drawer/FeedFilterDrawer.tsx:65]: this is about to get wrapped in another text inside DrawerHeader, right?

Comment (amendelsohn) [reply packages/mobile/src/components/feed-filter-drawer/FeedFilterDrawer.tsx:65]: should we take a different prop when you are passing a component vs plain text?

**Review by amendelsohn — APPROVED** [review]

File: packages/common/src/utils/linking.ts [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+  try {
+    let hostname = new URL(link).hostname
```
Comment (rickyrombo) [inline packages/common/src/utils/linking.ts:12]: Does react-native support `URL.hostname` now? Nice!

File: packages/mobile/src/screens/chat-screen/LinkPreview.tsx [hunk 6]
```diff
@@ -101,7 +102,7 @@ export const LinkPreview = ({
   const { onOpen: openLeavingAudiusModal } = useLeavingAudiusModal()
 
   const handlePress = useCallback(() => {
-    if (isAudiusUrl(href)) {
+    if (isAudiusUrl(href) || isAllowedExternalLink(href)) {
```
Comment (rickyrombo) [inline packages/mobile/src/screens/chat-screen/LinkPreview.tsx:105]: thanks!

File: packages/mobile/src/components/core/Hyperlink.tsx [hunk 7]
```diff
@@ -98,10 +102,10 @@ export const Hyperlink = (props: HyperlinkProps) => {
 
   const handlePress = useCallback(
     (url) => {
-      if (!isAudiusUrl(url)) {
-        openLeavingAudiusModal({ link: url })
-      } else {
+      if (isAudiusUrl(url) || isAllowedExternalLink(url)) {
```
Comment (rickyrombo) [inline packages/mobile/src/components/core/Hyperlink.tsx:105]: Thanks!

File: packages/mobile/src/components/drawer/Drawer.tsx [hunk 8]
```diff
@@ -111,7 +111,7 @@ export type DrawerProps = {
   /**
    * Header title if this drawer has a header
    */
-  title?: string
+  title?: ReactNode
```
Comment (rickyrombo) [inline packages/mobile/src/components/drawer/Drawer.tsx:114]: <3

**Review by rickyrombo — APPROVED** [review]

File: packages/mobile/src/components/core/Text.tsx (hunk 3 above)
Comment (dylanjeffers) [reply packages/mobile/src/components/core/Text.tsx:77]: completely agree

File: packages/mobile/src/components/feed-filter-drawer/FeedFilterDrawer.tsx (hunk 4 above)
Comment (dylanjeffers) [reply packages/mobile/src/components/feed-filter-drawer/FeedFilterDrawer.tsx:65]: i could def see that, i was thinking if you want to update the title text, you can just pass a text component ready to go, and it's okay that it's wrapped. really only applies to one drawer and ideally we should all have the same title. can see the idea of having a "renderTitle" thing that goes all the way down though.

File: packages/common/src/utils/linking.ts (hunk 5 above)
Comment (dylanjeffers) [reply packages/common/src/utils/linking.ts:13]: yeah it does thank heavens

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/dj-c-3060-fix-outbound-link-experienc

File: packages/mobile/src/components/feed-filter-drawer/FeedFilterDrawer.tsx (hunk 4 above)
Comment (amendelsohn) [reply packages/mobile/src/components/feed-filter-drawer/FeedFilterDrawer.tsx:65]: yeah either way is okay I think. As long as double wrapping the text uses the inner one I guess

**Pull request merged**